src/main/cpp/io/ArmIO.cpp

```diff
@@ -12,6 +13,19 @@ void ArmIO::SimulationPeriodic() {
 }
 
 bool ArmIO::ProcessIO() {
+    bool hasReadParameters = false;
+
+    // if we haven't read the parameters
+    if (!hasReadParameters) {
+        // update the variables
+        offset = RobotParams::GetParam("arm.offset", -330);
+        lift_limit = RobotParams::GetParam("arm.lift_limit", 100);
+        extend_limit = RobotParams::GetParam("arm.extend_limit", 100);
+        max_output = RobotParams::GetParam("arm.max_output", 0.2);
+
+        // raise the flag
+        hasReadParameters = true;
+    }
```

Comment:
No need to change now, but I feel like this should be in an Init function instead of hacked into ProcessIO

Reply:
yeah I was thinking about that, but call it where? RobotContainer? Oh, yeah, because RobotInit() gets called before RobotPeriodic(), so w could call an init() before periodic() starts. Yeah okay,